File: ApplicationCore/ComicBase.Core/Domain/Entities/Issue.cs
using ComicBase.Core.Domain.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace ComicBase.Core.Domain.Entities
{
    /// <summary>
    /// Entity for a single comic book issue. The publisher is linked via a navigation property.
    /// </summary>
    public class Issue : EntityBase
    {
        [Required]
        public string Title { get; set; }

        [Required]
        public int Year { get; set; }

        public int Volume { get; set; }

        [Required]
        public int IssueNumber { get; set; }

        public int PublisherId { get; set; }

        public virtual Publisher Publisher { get; set; }
    }
}

File: UserInterfaces/UI.Aurelia/ClientApp/app/components/Issue/Models/Issue.ts
import { Publisher } from "../../Publisher/Models/Publisher";

export class Issue {

    constructor(title: string, publisherId: number, publisher: Publisher, volume: number, issueNumber: number, year: number) {

        this.Title = title;
        this.PublisherId = publisherId;
        this.Publisher = publisher;
        this.Volume = volume;
        this.IssueNumber = issueNumber;
        this.Year = year;

    }

    public Id?: number;
    public Title: string;
    public PublisherId: number;
    public Publisher: Publisher;
    public Volume: number;
    public IssueNumber: number;
    public Year: number;

}